src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1731,6 +1733,31 @@ void reportPrimaryConstructorParameterShadowing(SimpleNameSyntax node, Symbol sy
             }
         }
 
+#nullable enable
+        internal void ReportFieldOrValueContextualKeywordConflicts(SyntaxNode syntax, string name, Symbol? symbol, BindingDiagnosticBag diagnostics)
```

Comment:
>symbol [](http://example.com/codeflow?start=107&length=6)

I'd consider a more specific name for this parameter so that it's clear when it's okay for a caller to pass `null`. For instance, `possibleAccessorValueParameter`.

Alternatively, I believe only one caller needs to pass an actual value, so it should be possible to use an optional boolean value "isAccessorValueParameter", or make the call conditional (`if (symbol is not SynthesizedAccessorValueParameterSymbol) { ReportFieldOrValueContextualKeywordConflicts(...); }`) and remove the parameter entirely.